src/context_store/storage/postgres.py

```
@@ -315,8 +342,41 @@ async def list_by_filter(self, filters: MemoryFilters) -> list[Memory]:
+        limit_clause = ""
+        limit_val = getattr(filters, "limit", None)
+        if limit_val is not None:
+            try:
+                limit_int = int(limit_val)
+                if limit_int >= 0:
+                    params.append(limit_int)
+                    limit_clause = f"LIMIT ${len(params)}"
+            except (ValueError, TypeError):
+                pass
+
+        sql = f"SELECT * FROM memories {where_clause} {order_clause} {limit_clause}".strip()
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)